src/sentry/utils/suspect_resolutions/get_suspect_resolutions.py

```diff
@@ -1,13 +1,21 @@
 from typing import Sequence
 
 from sentry.models import Activity, Group, GroupStatus
+from sentry.signals import issue_resolved
+from sentry.tasks.base import instrumented_task
 from sentry.utils.suspect_resolutions import analytics
 from sentry.utils.suspect_resolutions.commit_correlation import is_issue_commit_correlated
 from sentry.utils.suspect_resolutions.metric_correlation import is_issue_error_rate_correlated
 
 
-def get_suspect_resolutions(resolved_issue: Group) -> Sequence[Group]:
+@issue_resolved.connect(weak=False)
+def record_suspect_resolutions(organization_id, project, group, user, resolution_type, **kwargs):
+    get_suspect_resolutions.delay(group.id)
 
+
+@instrumented_task(name="sentry.tasks.get_suspect_resolutions", queue="get_suspect_resolutions")
+def get_suspect_resolutions(resolved_issue_id: int) -> Sequence[Group]:
```

Comment:
Just adding comments up here since I'm unable to comment lower down.

Looking through this code, I think it will be quite expensive to run this logic for every issue in a project. `is_issue_error_rate_correlated` makes a snuba query each time it's called, and `is_issue_commit_correlated` will make 4 postgres queries. In a larger project with say 1000 issues, this is a lot of querying to do per resolution. At the very least we're likely to hit snuba rate limits while attempting to run these queries.

I think there are a few things we can do here to help improve this:
 - We can limit the number of issues we fetch. This will make this less accurate, but it's probably necessary to make sure we don't cause production problems here. You could order these by `first_seen`, `last_seen` or some other criteria to try and get the most relevant issues. Probably limit to 50-100 issues.
 - The query in `is_issue_error_rate_correlated` could be batched to fetch data for all of the related issues at once. This will help us avoid being ratelimited by snuba.
 - `get_range` will return a lot of data for each group since we're bucketing by minute. Since we're always querying in a several hour window, we could probably safely roll up by 3600 here.
 - in `get_files_changed`
  -  I'd recommend using `.distinct()` on the `CommitFileChange` query to reduce the potential number of rows we return.
  - You can pass the `Releases` query directly to the `CommitFileChange` and it'll run as a subquery, so that we don't need to bring all releases into memory.
  - It's fairly hard to batch these queries well, but if we limit the number of issues this runs on it hopefully won't cause any problems. And if we have a feature flag, then if we do see problems we can disable this processing.

Reply:
Thanks Dan, implemented these changes in https://github.com/getsentry/sentry/pull/36871 - in terms of `get_range`, I think rolling up by 3600 will decrease the accuracy of the result since we’re looking at 5h of data, maybe 600 (10 min buckets) instead?

Reply:
600 should be ok. Snuba has a limit of 10k datapoints that it can return. If we have 100 issues and we're fetching 5 hours for each of them with 6 datapoints an hour then the total number of datapoints is 100 * 5 * 6 = 3000. As long as we don't fetch any more issues then we should be fine here.